src/adapters/scene.js

```diff
@@ -333,6 +328,11 @@ Scene.prototype.ensureMarkerIsVisible = function(poi, options) {
 };
 
 Scene.prototype.addMarker = function(poi) {
+  if (this.currentMarker?.latLon === poi.latLon) {
+    // Prevent setting the same poi multiple times
+    return this.currentMarker;
+  }
+
```

Comment:
This doesn't seem to have any effect, as `.latLon` is not a property of mapbox-gl `Marker`.
And this approach inspecting `this.currentMarker` does not look correct to me. What we need to check is that the map still contains a list of markers (stored by `SceneCategory`). They are distinct from `this.currentMarker`.

Here is a gif showing a problem with the proposed implementation:
![Peek 01-07-2020 18-23](https://user-images.githubusercontent.com/4726554/86268332-527d0600-bbc8-11ea-879d-8e5225711973.gif)

Reply:
You right, I updated the relevant code